src/display/editor/annotation_editor_layer.js

```diff
@@ -499,6 +504,36 @@ class AnnotationEditorLayer {
     return editor;
   }
 
+  /**
+   * Create and add a new editor.
+   */
+  addNewEditor() {
+    const { x, y, width, height } = this.div.getBoundingClientRect();
+    const tlX = Math.max(0, x);
+    const tlY = Math.max(0, y);
+    const brX = Math.min(window.innerWidth, x + width);
+    const brY = Math.min(window.innerHeight, y + height);
+    const centerX = (tlX + brX) / 2 - x;
+    const centerY = (tlY + brY) / 2 - y;
+    let [offsetX, offsetY] = [centerX, centerY];
+
+    switch (this.viewport.rotation) {
+      case 90:
+      case 270:
+        offsetX = centerY;
+        offsetY = centerX;
+        break;
+    }
```

Comment:
Nit: It occurred to me that all this could probably be simplify even further.
```suggestion
    const [offsetX, offsetY] =
      this.viewport.rotation % 180 === 90
        ? [centerY, centerX]
        : [centerX, centerY];
```